service/openapi/src/main/java/org/dotwebstack/framework/service/openapi/OpenApiConfiguration.java

```diff
@@ -34,60 +38,89 @@
       ResponseContextValidator responseContextValidator) {
     this.graphQl = graphQl;
-    this.typeDefinitionRegistry = typeDefinitionRegistry;
     this.paramHandlerRouter = paramHandlerRouter;
-    this.graphQlFieldBuilder = new GraphQlFieldBuilder(this.typeDefinitionRegistry);
     this.responseMapper = responseMapper;
     this.responseContextValidator = responseContextValidator;
+    this.queryFieldHelper = QueryFieldHelper.builder()
+        .typeDefinitionRegistry(typeDefinitionRegistry)
+        .graphQlFieldBuilder(new GraphQlFieldBuilder(typeDefinitionRegistry))
+        .build();
   }
 
   @Bean
   public RouterFunction<ServerResponse> route(@NonNull OpenAPI openApi) {
     RouterFunctions.Builder routerFunctions = RouterFunctions.route();
-    QueryFieldHelper queryFieldHelper = QueryFieldHelper.builder()
-        .typeDefinitionRegistry(this.typeDefinitionRegistry)
-        .graphQlFieldBuilder(this.graphQlFieldBuilder)
-        .build();
+
     ResponseTemplateBuilder responseTemplateBuilder = ResponseTemplateBuilder.builder()
         .openApi(openApi)
         .build();
+
     openApi.getPaths()
-        .forEach((name, path) -> {
-          if (Objects.nonNull(path.getGet())) {
-            routerFunctions.add(toRouterFunction(responseTemplateBuilder, name,
-                queryFieldHelper.resolveGraphQlField(path.getGet()), "get", path.getGet(), GET(name)));
-          }
-          if (Objects.nonNull(path.getPost())) {
-            routerFunctions.add(toRouterFunction(responseTemplateBuilder, name,
-                queryFieldHelper.resolveGraphQlField(path.getPost()), "post", path.getPost(), POST(name)));
-          }
-        });
+        .forEach((name, path) -> Stream.of(path)
+            .flatMap(p -> getHttpMethodOperations(path, name).stream())
+            .flatMap(httpMethodOperation -> toRouterFunctions(responseTemplateBuilder, httpMethodOperation).stream())
+            .forEach(routerFunctions::add));
+
     return routerFunctions.build();
   }
 
-  protected RouterFunction<ServerResponse> toRouterFunction(ResponseTemplateBuilder responseTemplateBuilder,
-      String path, GraphQlField graphQlField, String methodName, Operation operation,
-      RequestPredicate requestPredicate) {
-    List<ResponseTemplate> responseTemplates =
-        responseTemplateBuilder.buildResponseTemplates(path, methodName, operation);
+  private List<HttpMethodOperation> getHttpMethodOperations(PathItem pathItem, String name) {
+    HttpMethodOperation.HttpMethodOperationBuilder builder = HttpMethodOperation.builder()
+        .name(name);
+
+    List<HttpMethodOperation> result = new ArrayList<>();
+
+    if (Objects.nonNull(pathItem.getGet())) {
+      result.add(builder.httpMethod(HttpMethod.GET)
+          .operation(pathItem.getGet())
+          .build());
+    }
+    if (Objects.nonNull(pathItem.getPost())) {
+      result.add(builder.httpMethod(HttpMethod.POST)
+          .operation(pathItem.getGet())
```

Comment:
```suggestion
          .operation(pathItem.getPost())
```
Geen test die hierop had moeten falen?

Reply:
Lastig te testen, bug wel gefixed.